tests/BrowserKitConfig.php

```diff
@@ -35,4 +35,29 @@ protected function supportsJs()
     {
         return false;
     }
+
+    public function skipMessage($testCase, $test)
+    {
+        if ('Behat\Mink\Tests\Driver\Form\Html5Test' === $testCase && 'testHtml5FormAction' === $test && version_compare($this->getDomCrawlerVersion(), '3.3.0.0-dev', '<')) {
+            return 'Symfony DomCrawler < 3.3 does not support the formAction attribute.';
+        }
+
+        return parent::skipMessage($testCase, $test);
+    }
+
+    private function getDomCrawlerVersion()
+    {
+        $installedFile = __DIR__.'/../vendor/composer/installed.json';
```

Comment:
yeah, relying on knowing the binary, then calling it in a cross-platform way and parsing the output (which is not guaranteed to stay exactly the same across composer versions) is the alternative. And IMO, it is uglier. Especially when you consider that we need to deal with both ``symfony/dom-crawler`` and ``symfony/symfony`` (as some Travis jobs are requiring the whole Symfony package to force using the LTS version of *all* components), and that we need the *normalized* version (suitable for usage in version_compare, due to version_compare not dealing well with all semver cases), which is **not** part of the ``composer show`` output (forcing us to depend on composer too to have their normalization logic).

I may not have done this code reading the composer metadata directly for production code (it would become more complex to account for the different config that projects may use). But this is only for our testsuite, so we know that we don't use crazy composer config (we control the composer.json) and it is good enough IMO